sdk/go.mod

```diff
@@ -46,6 +46,7 @@ require (
 	github.com/pierrec/lz4 v2.6.1+incompatible
 	github.com/robfig/cron/v3 v3.0.1
 	github.com/ryanuber/go-glob v1.0.0
+	github.com/snowflakedb/gosnowflake v1.13.2
```

Comment:
hmm, I don't love the idea of importing snowflake into the SDK. This would mean that any consumers of the Vault SDK will also import snowflake. I wonder if we could avoid this by having the Snowflake plugin override the ConnectionProducer's Connection() function?

Reply:
I just checked adds 30MB to the SDK. Which will double the binary size of Couchbase db plugin for example:

On main:
```
make dev
...
-rwxr-xr-x@ 1 jmf  staff    32M May  1 13:04 vault-plugin-database-couchbase
```

With this change
```
go get github.com/hashicorp/vault/sdk@abb4641b409792a9d0a503b12bdc0c3a21f3360b
go mod tidy

make dev
...
-rwxr-xr-x@ 1 jmf  staff    63M May  1 13:03 vault-plugin-database-couchbase
```